File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraTreeList.Nodes.Operations;
using DevExpress.XtraTreeList.Columns;


namespace WindowsApplication385
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		private DevExpress.XtraTreeList.TreeList treeList1;
		private DevExpress.XtraTreeList.Columns.TreeListColumn treeListColumn1;
		private DevExpress.XtraTreeList.Columns.TreeListColumn treeListColumn2;
		private DevExpress.XtraEditors.Repository.RepositoryItemCheckEdit repositoryItemCheckEdit1;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.treeList1 = new DevExpress.XtraTreeList.TreeList();
            this.treeListColumn1 = new DevExpress.XtraTreeList.Columns.TreeListColumn();
            this.repositoryItemCheckEdit1 = new DevExpress.XtraEditors.Repository.RepositoryItemCheckEdit();
            this.treeListColumn2 = new DevExpress.XtraTreeList.Columns.TreeListColumn();
            ((System.ComponentModel.ISupportInitialize)(this.treeList1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.repositoryItemCheckEdit1)).BeginInit();
            this.SuspendLayout();
            // 
            // treeList1
            // 
            this.treeList1.Columns.AddRange(new DevExpress.XtraTreeList.Columns.TreeListColumn[] {
            this.treeListColumn1,
            this.treeListColumn2});
            this.treeList1.Dock = System.Windows.Forms.DockStyle.Top;
            this.treeList1.Location = new System.Drawing.Point(0, 0);
            this.treeList1.Name = "treeList1";
            this.treeList1.BeginUnboundLoad();
            this.treeList1.AppendNode(new object[] {
            false,
            "1"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "2"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "21"}, 1);
            this.treeList1.AppendNode(new object[] {
            false,
            "211"}, 2);
            this.treeList1.AppendNode(new object[] {
            false,
            "212"}, 2);
            this.treeList1.AppendNode(new object[] {
            false,
            "213"}, 2);
            this.treeList1.AppendNode(new object[] {
            false,
            "214"}, 2);
            this.treeList1.AppendNode(new object[] {
            false,
            "22"}, 1);
            this.treeList1.AppendNode(new object[] {
            false,
            "23"}, 1);
            this.treeList1.AppendNode(new object[] {
            false,
            "24"}, 1);
            this.treeList1.AppendNode(new object[] {
            false,
            "3"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "4"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "5"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "6"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "7"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "8"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "9"}, -1);
            this.treeList1.AppendNode(new object[] {
            false,
            "10"}, -1);
            this.treeList1.EndUnboundLoad();
            this.treeList1.RepositoryItems.AddRange(new DevExpress.XtraEditors.Repository.RepositoryItem[] {
            this.repositoryItemCheckEdit1});
            this.treeList1.Size = new System.Drawing.Size(592, 400);
            this.treeList1.TabIndex = 0;
            // 
            // treeListColumn1
            // 
            this.treeListColumn1.Caption = "Column1";
            this.treeListColumn1.ColumnEdit = this.repositoryItemCheckEdit1;
            this.treeListColumn1.FieldName = "treeListColumn1";
            this.treeListColumn1.MinWidth = 56;
            this.treeListColumn1.Name = "treeListColumn1";
            this.treeListColumn1.Visible = true;
            this.treeListColumn1.VisibleIndex = 0;
            // 
            // repositoryItemCheckEdit1
            // 
            this.repositoryItemCheckEdit1.AutoHeight = false;
            this.repositoryItemCheckEdit1.Name = "repositoryItemCheckEdit1";
            this.repositoryItemCheckEdit1.CheckedChanged += new System.EventHandler(this.repositoryItemCheckEdit1_CheckedChanged);
            // 
            // treeListColumn2
            // 
            this.treeListColumn2.Caption = "Column2";
            this.treeListColumn2.FieldName = "treeListColumn2";
            this.treeListColumn2.Name = "treeListColumn2";
            this.treeListColumn2.Visible = true;
            this.treeListColumn2.VisibleIndex = 1;
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(592, 527);
            this.Controls.Add(this.treeList1);
            this.Name = "Form1";
            this.Text = "How to change the field value of all child nodes of a certain node";
            ((System.ComponentModel.ISupportInitialize)(this.treeList1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.repositoryItemCheckEdit1)).EndInit();
            this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		public class TreeListChangeChildNodesOperation : TreeListOperation 
		{
			private TreeListColumn column;
			private bool state;
			private TreeListNode parentNode;
			public TreeListChangeChildNodesOperation(TreeListColumn column, TreeListNode node, bool state) 
			{ 
				this.column = column;
				this.state = state;
				this.parentNode = node;
			}

			public override void Execute(TreeListNode node) 
			{
				if (node.HasAsParent(parentNode))
					node.SetValue(column, state);
				else
					if (parentNode.HasAsParent(node))
						if (state)
							node[column] = state;
						else
							node[column] = null;
			}				

			public override bool NeedsVisitChildren(TreeListNode node) { return true; }
		}

		private void repositoryItemCheckEdit1_CheckedChanged(object sender, System.EventArgs e)
		{
			treeList1.PostEditor();
			TreeListNode node = treeList1.FocusedNode;
			TreeListChangeChildNodesOperation op = new TreeListChangeChildNodesOperation(treeListColumn1, node, (sender as DevExpress.XtraEditors.CheckEdit).Checked);
			treeList1.NodesIterator.DoOperation(op);			
		}
	}
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.Nodes
Imports DevExpress.XtraTreeList.Nodes.Operations
Imports DevExpress.XtraTreeList.Columns


Namespace WindowsApplication385
	''' <summary>
	''' Summary description for Form1.
	''' </summary>
	Public Class Form1
		Inherits System.Windows.Forms.Form
		Private treeList1 As DevExpress.XtraTreeList.TreeList
		Private treeListColumn1 As DevExpress.XtraTreeList.Columns.TreeListColumn
		Private treeListColumn2 As DevExpress.XtraTreeList.Columns.TreeListColumn
		Private WithEvents repositoryItemCheckEdit1 As DevExpress.XtraEditors.Repository.RepositoryItemCheckEdit
		''' <summary>
		''' Required designer variable.
		''' </summary>
		Private components As System.ComponentModel.Container = Nothing

		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()

			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If components IsNot Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.treeList1 = New DevExpress.XtraTreeList.TreeList()
			Me.treeListColumn1 = New DevExpress.XtraTreeList.Columns.TreeListColumn()
			Me.repositoryItemCheckEdit1 = New DevExpress.XtraEditors.Repository.RepositoryItemCheckEdit()
			Me.treeListColumn2 = New DevExpress.XtraTreeList.Columns.TreeListColumn()
			CType(Me.treeList1, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.repositoryItemCheckEdit1, System.ComponentModel.ISupportInitialize).BeginInit()
			Me.SuspendLayout()
			' 
			' treeList1
			' 
			Me.treeList1.Columns.AddRange(New DevExpress.XtraTreeList.Columns.TreeListColumn() { Me.treeListColumn1, Me.treeListColumn2})
			Me.treeList1.Dock = System.Windows.Forms.DockStyle.Top
			Me.treeList1.Location = New System.Drawing.Point(0, 0)
			Me.treeList1.Name = "treeList1"
			Me.treeList1.BeginUnboundLoad()
			Me.treeList1.AppendNode(New Object() { False, "1"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "2"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "21"}, 1)
			Me.treeList1.AppendNode(New Object() { False, "211"}, 2)
			Me.treeList1.AppendNode(New Object() { False, "212"}, 2)
			Me.treeList1.AppendNode(New Object() { False, "213"}, 2)
			Me.treeList1.AppendNode(New Object() { False, "214"}, 2)
			Me.treeList1.AppendNode(New Object() { False, "22"}, 1)
			Me.treeList1.AppendNode(New Object() { False, "23"}, 1)
			Me.treeList1.AppendNode(New Object() { False, "24"}, 1)
			Me.treeList1.AppendNode(New Object() { False, "3"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "4"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "5"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "6"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "7"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "8"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "9"}, -1)
			Me.treeList1.AppendNode(New Object() { False, "10"}, -1)
			Me.treeList1.EndUnboundLoad()
			Me.treeList1.RepositoryItems.AddRange(New DevExpress.XtraEditors.Repository.RepositoryItem() { Me.repositoryItemCheckEdit1})
			Me.treeList1.Size = New System.Drawing.Size(592, 400)
			Me.treeList1.TabIndex = 0
			' 
			' treeListColumn1
			' 
			Me.treeListColumn1.Caption = "Column1"
			Me.treeListColumn1.ColumnEdit = Me.repositoryItemCheckEdit1
			Me.treeListColumn1.FieldName = "treeListColumn1"
			Me.treeListColumn1.MinWidth = 56
			Me.treeListColumn1.Name = "treeListColumn1"
			Me.treeListColumn1.Visible = True
			Me.treeListColumn1.VisibleIndex = 0
			' 
			' repositoryItemCheckEdit1
			' 
			Me.repositoryItemCheckEdit1.AutoHeight = False
			Me.repositoryItemCheckEdit1.Name = "repositoryItemCheckEdit1"
'			Me.repositoryItemCheckEdit1.CheckedChanged += New System.EventHandler(Me.repositoryItemCheckEdit1_CheckedChanged);
			' 
			' treeListColumn2
			' 
			Me.treeListColumn2.Caption = "Column2"
			Me.treeListColumn2.FieldName = "treeListColumn2"
			Me.treeListColumn2.Name = "treeListColumn2"
			Me.treeListColumn2.Visible = True
			Me.treeListColumn2.VisibleIndex = 1
			' 
			' Form1
			' 
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
			Me.ClientSize = New System.Drawing.Size(592, 527)
			Me.Controls.Add(Me.treeList1)
			Me.Name = "Form1"
			Me.Text = "How to change the field value of all child nodes of a certain node"
			CType(Me.treeList1, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.repositoryItemCheckEdit1, System.ComponentModel.ISupportInitialize).EndInit()
			Me.ResumeLayout(False)

		End Sub
		#End Region

		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		<STAThread> _
		Shared Sub Main()
			Application.Run(New Form1())
		End Sub

		Public Class TreeListChangeChildNodesOperation
			Inherits TreeListOperation
			Private column As TreeListColumn
			Private state As Boolean
			Private parentNode As TreeListNode
			Public Sub New(ByVal column As TreeListColumn, ByVal node As TreeListNode, ByVal state As Boolean)
				Me.column = column
				Me.state = state
				Me.parentNode = node
			End Sub

			Public Overrides Sub Execute(ByVal node As TreeListNode)
				If node.HasAsParent(parentNode) Then
					node.SetValue(column, state)
				Else
					If parentNode.HasAsParent(node) Then
						If state Then
							node(column) = state
						Else
							node(column) = Nothing
						End If
					End If
				End If
			End Sub

			Public Overrides Function NeedsVisitChildren(ByVal node As TreeListNode) As Boolean
				Return True
			End Function
		End Class

		Private Sub repositoryItemCheckEdit1_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles repositoryItemCheckEdit1.CheckedChanged
			treeList1.PostEditor()
			Dim node As TreeListNode = treeList1.FocusedNode
			Dim op As New TreeListChangeChildNodesOperation(treeListColumn1, node, (TryCast(sender, DevExpress.XtraEditors.CheckEdit)).Checked)
			treeList1.NodesIterator.DoOperation(op)
		End Sub
	End Class
End Namespace
